packages/yoga/src/Theme/Provider/Provider.jsx

```diff
@@ -4,15 +4,36 @@ import PropTypes from 'prop-types';
 import * as tokens from '@gympass/yoga-tokens';
 import yogaTheme from '../theme';
 
-const getTheme = ({ locale }) => {
+/**
+ * Gets resolved theme to inject in Yoga
+ *
+ * @param {{locale: string, theme: function}} themeParameters information about the theme
+ * @param {string} themeParameters.locale custom locale to get colors from (if it has a different scheme)
+ * @param {function} themeParameters.theme returns a custom theme object to replace existing tokens
+ * @returns {Object.<string, string | number>} theme object with tokens sent to styled-components' <ThemeProvider />
+ */
+const getTheme = ({ locale, theme }) => {
+  const isThemeInvalid = theme ? !('isFromThemeBuilder' in theme) : false;
+
+  if (isThemeInvalid) {
+    throw new TypeError(
+      `Invalid 'customTheming' parameter. Expected one generated by createTheme() function.
+    Did you sent a function directly to <ThemeProvider />?
+    
+    const getTheme = ({ locale, theme }) => {
```

Comment:
I don't think that this error is helpful, since the end-user does not work with this function he doesn't know what `getTheme` means. Maybe we should say something like "_The Provider prop has an invalid type_" or something like that.

What do you think?

Reply:
The stack trace would already point to the internal source, I believe that `'customTheming'` isn't clear enough (changing it rn), but providing the engineer more context after the original message is good so one may know where to see the implementation in order to understand it better.

The "helpful" information (the one expected to quickly solve the issue is the first one. Makes sense to you?

cc/ @ggdaltoso